exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillCursor.java

```diff
@@ -333,8 +368,14 @@ void close() {
     final int batchQueueThrottlingThreshold =
         client.getConfig().getInt(
             ExecConstants.JDBC_BATCH_QUEUE_THROTTLING_THRESHOLD );
-    resultsListener = new ResultsListener(batchQueueThrottlingThreshold);
+    resultsListener = new ResultsListener(this, batchQueueThrottlingThreshold);
     currentBatchHolder = new RecordBatchLoader(client.getAllocator());
+    try {
+      setTimeout(this.statement.getQueryTimeout());
+    } catch (SQLException e) {
+      // Printing any unexpected SQLException stack trace
+      e.printStackTrace();
```

Comment:
two choices here:
- we don't think it's important if we cannot get the value, so we should log it properly and not simply dump the exception
- we think this is important, and we propagate the exception to the caller

(I think it is important: the most likely reason why we could not get the value if that the statement was closed, and we should probably notify the user about it).

Reply:
I agree. Thankfully, the _caller_ does handle any thrown `SQLException`s, so I'm going to pass this off to that. IMO, I don't think we'll have an issue because the `Statement.setQueryTimeout()` would have handled any corner cases before this is invoked via `execute()`